fuzz/ocsp_parse_url.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/mem.h>
+#include <openssl/ocsp.h>
+#include <string>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len) {
+  std::string input(reinterpret_cast<const char *>(buf),
+                    std::next(reinterpret_cast<const char *>(buf), len));
+  char *host = nullptr;
+  char *path = nullptr;
+  char *port = nullptr;
+  int is_ssl;
+
+  if (OCSP_parse_url(input.c_str(), &host, &port, &path, &is_ssl)) {
+    // Check that parameters have been assigned values.
+    if (!host && !path && !port && (is_ssl != 1 && is_ssl != 0)) {
+      return 1;
+    }
```

Comment:
Is the parsing all or nothing? Could we end up parsing a part of the URL?

Reply:
Yeah it's all or nothing. We have some test cases covering certain empty components.
* https://github.com/aws/aws-lc/blob/ocsp-request/crypto/ocsp/ocsp_test.cc#L1116-L1117